rasa/core/policies/ted_policy.py

```diff
@@ -1092,36 +1095,84 @@ def _compute_dialogue_indices(
             ).values
         ]
 
-    def _create_all_labels_embed(self) -> Tuple[tf.Tensor, tf.Tensor]:
+    def _create_all_labels_embed_for_training(
+        self, number_of_random_labels: int
+    ) -> Tuple[tf.Tensor, tf.Tensor]:
+
         all_label_ids = self.tf_label_data[LABEL_KEY][LABEL_SUB_KEY][0]
-        # labels cannot have all features "fake"
-        all_labels_encoded = {}
+
+        if number_of_random_labels != -1:
+
+            # First, sample a fixed number of random labels ids
+            total_num_label_ids = all_label_ids.shape[0]
```

Comment:
I think, it should be
```suggestion
            total_num_label_ids = tf.shape(all_label_ids)[0]
```
not sure if `all_label_ids` has static shape, but maybe fine if it is

Reply:
`all_label_ids` has static shape, also using `tf.shape` will return a dynamic shape which cannot be used as an argument to `candidate_sampler`.
In summary, this should work as expected.

Reply:
`all_label_ids` has static shape, also using `tf.shape` will return a dynamic shape which cannot be used as an argument to `candidate_sampler`.
In summary, this should work as expected.